### ISSUE TITLE
Binding on a multilevel path doesn't seem to real time update

### ISSUE THREAD

**Issue by thebrauer** [Starcounter/Starcounter#3704]
We have a model that looks like this:
http://screencast.com/t/C3XgQOzsa

At first we only had Name, Address and Metadata with children. When we updated the value for Errors and we had a binding like this inside a repeat: {{menuitem.Metadata.2.Value}} the binding works but the value only updates in the browser after a refresh. 

But when we added the properties directly on the menu item and bound like {{menuitem.Errors}} it updates correctly in real time.

Is there maybe a problem with the number in the path?

**Comment by miyconst** [comment]
@thebrauer could you please provide some more info of what you are trying to do? The HTML template might help.

I don't understand why would you need a binding by an array item number like this `menuitem.Metadata.2.Value`.

**Comment by careri** [comment]
We're using partials to update part of a page. Each partial page is bound to a menu item.
The menu item can then display additional information based on metadata.

In this case it's the menu-item that takes us to the workstations page. This page aggregates all known errors from all known workstations into a number that is displayed as an angry digit in the lower left corner of the menu item.

Other pages might display other metadata.

We have a deadline for our project, so we did a hardcoded QAD binding using the indexes of the metadata. This was not the final solution since it should support additional metadata types.

But enough about that, the question is. Why doesn't the bound value including array-index update? The first binding works, but updates doesn't trigger the bound value.

**Comment by miyconst** [comment]
It looks like this is a Polymer issue. Please see my jsfiddle - https://jsfiddle.net/qpf1235c/.

**Comment by warpech** [comment]
Yeah, looks like this should be reported to Polymer. Thanks @miyconst for investigating this

**Comment by dan31** [comment]
Hm, interesting one. @warpech do you know if this has been moved on the Polymer side?

**Comment by warpech** [comment]
I don't think so. In our sample apps we are working around this problem using string keys, so instead of `menuitem.Metadata.2.Value` we do `menuitem.Metadata.Row2.Value`.

But as @miyconst pointed out, this is a very rare problem to need to explicitly bind to a specific array item.

Polymer does not fix any issues in Polymer 1.x and this problem is likely to be gone in Polymer 2. We will be upgrading to Polymer 2 in Q2 2017.

**Comment by tomalec** [comment]
Polymer 1.x is known to have plenty of issues around bindings to arrays and indexes/keys resolutions. They claim that Polymer 2.0 has redesigned data binding that should solve those,  but I didn't have time to check all of them.
